contents/usr/share/man/man1/textidote.1
.TH TEXTIDOTE 1 "2021-03-31" "GNU" "TeXtidote Abbreviated User's Manual"
.SH NAME
textidote \- proof reader for LaTeX and Markdown files
.SH SYNOPSIS
.B textidote
.I "[options] file1 [file2 ...]"
.br
.SH "DESCRIPTION"
.B TeXtidote
is a proof reader, spelling and grammar checker for LaTeX documents. The process
of checking LaTeX documents (especially grammar) is far from simple. Since LaTeX
documents contain special commands and keywords (the so-called "markup") that
are not part of the "real" text, you cannot run a grammar checker directly on
these files: it cannot tell the difference between markup and text. The other
option is to remove all this markup, leaving only the "clear" text; however,
when a grammar tool points to a problem at a specific line in this clear text,
it becomes hard to retrace that location in the original LaTeX file.
.PP
TeXtidote solves this problem; it can read your original LaTeX file and
perform various sanity checks on it: for example, making sure that every
figure is referenced in the text, enforcing the correct capitalization of
titles, etc. In addition, TeXtidote can remove markup from the file and send
it to the Language Tool library, which performs a verification of
.B both spelling and grammar
in a dozen languages.
What is unique to TeXtidote is that it keeps track of the relative position of
words between the original and the "clean" text. This means that it can
translate the messages from Language Tool back to their proper location
.B directly in your source file.
.PP
TeXtidote also supports spelling and grammar checking of files in the
Markdown format. Simply pass a Markdown file instead of a LaTeX file.
.SS OPTIONS
When launched with no option, TeXtidote analyzes the input file(s) and produces
a report of all errors and warnings found; this report is printed on the
standard output. If no input file is specified, TeXtidote reads from the
standard input.
.TP
\fB--output\fP
Outputs the result of the analysis to the standard output. Possible values are
\fIplain\fR (default), \fIhtml\fR, \fIjson\fR or \fIsingleline\fR.
In the case of the HTML report, the output shows the input LaTeX source file,
and the errors found are highlighted directly in the source at the corresponding
location. This report can be saved to a file and opened in a web browser.
.TP
\fB--check\fR \fIlang\fR
Run spelling and grammar checks on language \fIlang\fR. The parameter \fIlang\fR
is a two-letter code indicating the language of the input text. The code
\fIen\fR stands for English; TeXtidote also supports \fIfr\fR (French), \fInl\fR
(Dutch), \fIde\fR (German), \fIes\fR (Spanish) and \fIpt\fR (Portuguese).
.TP
\fB--ignore\fR \fIlist\fR
Ignore advice produced by some rules. The list is a comma-separated list of
rule IDs (with no spaces). IDs can contain a star as a wildcard charater.
.TP
\fB--dict\fP \fIfile\fR
Use \fIfile\fR as a dictionary. This option only has an effect when used with
\fB--check\fR. Parameter \fIfile\fR should point to a text file containing a
list of words that will be ignored by the spell checker.
.TP
\fB--replace\fP \fIfile\fR
Apply find/replace patterns described in \fIfile\fR. This file should contain
one pattern per line, with the search pattern separated by a tab from the
replacement pattern. Both search and replace can be regular expressions.
.TP
\fB--remove\fP \fIenvs\fR
Remove environments provided in the comma-separated list \fIenvs\fR.
.TP
\fB--remove-macros\fP \fImacs\fR
Remove user-defined macros provided in the comma-separated list \fImacs\fR.
.TP
\fB--clean\fP
Remove LaTeX markup from the file and print the result to the
standard output.
.TP
\fB--read-all\fP
By default, TeXtidote ignores everything before the \fI\\begin{document}\fR
command. Use this option to force TeXtidote to read all the file. This is
useful if you are analyzing a file that is a fragment of a larger document.
.TP
\fB--map\fP \fIfile\fR
Output correspondences between character ranges in \fIfile\fR. This option only
has an effect when used with \fB--detex\fR. TeXtidote keeps a mapping between
character ranges in the "detexed" file, and the same character ranges in the
original LaTeX document. This parameter can output the mapping to a text file.
.TP
\fB--type\fP \fIt\fR
Tells TeXtidote that the input file type is t. Supported values are \fItex\fR
(LaTeX) and \fImd\fR (Markdown).
.TP
\fB--no-color\fP
Disable the use of ANSI escape codes when printing reports to the standard
output (only has an effect when the \fB--html\fP option is not used).
.TP
\fB--no-config\fP
Ignore the default parameters found in the \fI.textidote\fR file (if it
exists).
.TP
\fB--help\fP
Show the command line usage.
.TP
\fB--version\fP
Show version number.
.SH SUPPORT
Support for TeXtidote can be found on the GitHub repository at
<https://github.com/sylvainhalle/textidote>.
Bug reports and suggestions should be submitted via the issue tracker.
.SH AUTHOR
This manual page was written by Sylvain Hallé <shalle@acm.org>.


Completions/textidote.bash
# Bash completion script for TeXtidote
# Based on this template:
# https://debian-administration.org/article/317/An_introduction_to_bash_completion_part_2

beginswith() 
{ 
	case $2 in 
		"$1"*) true;; 
		*) false;; 
	esac; 
}

# Found here: https://superuser.com/a/564776
completeFiles()
{
	local IFS=$'\n'
    local LASTCHAR=' '
    COMPREPLY=($(compgen -o plusdirs -f -- "${COMP_WORDS[COMP_CWORD]}"))
    if [ ${#COMPREPLY[@]} = 1 ]; then
        [ -d "$COMPREPLY" ] && LASTCHAR=/
        COMPREPLY=$(printf %q%s "$COMPREPLY" "$LASTCHAR")
    else
        for ((i=0; i < ${#COMPREPLY[@]}; i++)); do
            [ -d "${COMPREPLY[$i]}" ] && COMPREPLY[$i]=${COMPREPLY[$i]}/
        done
    fi
}

_textidote()
{
	local cur prev opts base
	COMPREPLY=()
	cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    
    #
    #  The basic options we'll complete.
    #
    opts="--check --clean --dict --help --ignore --languagemodel --map --no-color --no-config --output --quiet --read-all --remove --remove-macros --replace --type"
    
    #
    #  Complete the arguments to some of the basic commands.
    #
    case "${prev}" in
        --check)
            local langs="de de_AT de_CH de_DE en en_CA en_UK en_US es fr nl pt"
            COMPREPLY=( $(compgen -W "${langs}" -- ${cur}) )
            return 0
            ;;
        --dict)
            completeFiles
            return 0
            ;;
        --map)
            completeFiles
            return 0
            ;;
        --output)
            local methods="html json plain singleline"
            COMPREPLY=( $(compgen -W "${methods}" -- ${cur}) )
            return 0
            ;;
        --replace)
            completeFiles
            return 0
            ;;
        --type)
            local types="tex md"
            COMPREPLY=( $(compgen -W "${types}" -- ${cur}) )
            return 0
            ;;
        *)
            ;;
    esac
    
    # If we are writing an option, complete from option list
    if beginswith "-" $cur; then
    	COMPREPLY=($(compgen -W "${opts}" -- ${cur}))
    	return 0
    fi

   # Reply with list of TeX files
   completeFiles
   return 0
}

_textidote_zsh()
{
	compadd --check --clean --dict --help --ignore --languagemodel --map --no-color --no-config --output --quiet --read-all --remove --remove-macros --replace --type
}

# Register the goto completions.
if [ -n "${BASH_VERSION}" ]; then
  if ! [[ $(uname -s) =~ Darwin* ]]; then
    complete -o filenames -F _textidote textidote
  else
    complete -F _textidote textidote
  fi
elif [ -n "${ZSH_VERSION}" ]; then
  compdef_textidote_zsh textidote
else
  echo "Unsupported shell."
  exit 1
fi

# :mode=bash: